src/Illuminate/Database/Query/Builder.php

```diff
@@ -4356,6 +4392,11 @@ public function __call($method, $parameters)
             return $this->macroCall($method, $parameters);
         }
 
+        $pattern = '/^(or)?(Where)(Any|All|None)(.+)$/i';
+        if (preg_match($pattern, $method, $matches) === 1) {
+            return $this->dynamicWhereColumns($matches, $parameters);
+        }
+
         if (str_starts_with($method, 'where')) {
```

Comment:
I'm curious is there a precedence of why dynamic where does not support `orWhere`; and should this new addition follows that to do away without `orWhere` or add the `orWhere` to the dynamic where condition.

Reply:
I don't think there is a precedence, the dynamic `where` functionality was added over 10 years ago. I guess support for orWhere for the dynamic columns could be a nice feature as well 😉

https://github.com/laravel/framework/commit/3a5ed243#diff-2ed94a0ea151404a12f3c0d52ae9fb5742348578ec4a8ff79d079fa598ff145d